### PR TITLE
Minor RPI4 install changes

### PR DESCRIPTION
**Pull request by recursinging**
These changes handle the issues I had setting this up on a Raspberry Pi 4 B, both during the initial setup as well as after subsequent booting.

### CODE REVIEW

File: scripts/raspberryPiSetup.sh [hunk 1]
```diff
@@ -43,17 +43,17 @@ echo "Extracted device major number: ${USB_DEVICE_MAJOR}"
...
 # write these to setupEnvVars.sh
 echo "# Appended from rapsberryPiSetup.sh:" >> setupEnvVars.sh
-echo "DAISYHAT_REPO=${REPO_NAME}" >> setupEnvVars.sh
-echo "DAISYHAT_USER=${USER_NAME}" >> setupEnvVars.sh
-echo "DAISYHAT_GH_TOKEN=${GH_TOKEN}" >> setupEnvVars.sh
-echo "DAISYHAT_DEVICE_MAJOR=${USB_DEVICE_MAJOR}" >> setupEnvVars.sh
+echo "export DAISYHAT_REPO=${REPO_NAME}" >> setupEnvVars.sh
+echo "export DAISYHAT_USER=${USER_NAME}" >> setupEnvVars.sh
+echo "export DAISYHAT_GH_TOKEN=${GH_TOKEN}" >> setupEnvVars.sh
+echo "export DAISYHAT_DEVICE_MAJOR=${USB_DEVICE_MAJOR}" >> setupEnvVars.sh
```
Comment (TheSlowGrowth) [inline scripts/raspberryPiSetup.sh:49]: Interesting. I had it like this earlier and was facing issues with this. Not sure what the problem was, tbh. I'm not a bash expert but it should work either way. Do you know why it didn't work for you?

File: scripts/raspberryPiSetup.sh [hunk 2]
```diff
@@ -43,17 +43,17 @@ echo "Extracted device major number: ${USB_DEVICE_MAJOR}"
...
 source ./setupEnvVars.sh
 # include this file with every bash session
-echo "# include daisyhat environment if it exists" >> $HOME/.bash_profile
-echo "if [ -f $HOME/daisyHat/setupEnvVars.sh ]; then" >> $HOME/.bash_profile
-echo "    source $HOME/daisyHat/setupEnvVars.sh" >> $HOME/.bash_profile
-echo "fi" >> $HOME/.bash_profile
+echo "# include daisyhat environment if it exists" >> $HOME/.profile
+echo "if [ -f $HOME/daisyHat/setupEnvVars.sh ]; then" >> $HOME/.profile
+echo "    source $HOME/daisyHat/setupEnvVars.sh" >> $HOME/.profile
+echo "fi" >> $HOME/.profile
```
Comment (TheSlowGrowth) [inline scripts/raspberryPiSetup.sh:56]: Oh, why did `.bash_profile` not work for you?

File: docker/Dockerfile [hunk 3]
```diff
@@ -36,7 +36,7 @@ RUN chown -R actions ~actions && /home/actions/actions-runner/bin/installdepende
 RUN apt-get update && apt-get install -y udev usbutils
 RUN mkdir -p /etc/udev/rules.d/
 COPY 49-stlinkv3.rules /etc/udev/rules.d/49-stlinkv3.rules
-RUN service udev restart || 
+RUN service udev restart
```
Comment (TheSlowGrowth) [inline docker/Dockerfile:39]: To my understanding the `||` should proceed with the Dockerfile build even if the restart fails. I guess it's okay to fail the build.

File: scripts/setupEnvVars.sh [hunk 4]
```diff
@@ -1,11 +1,12 @@
...
-# DAISYHAT_DEVICE_MAJOR=189 # major device number of devices to passthrough to the container
-# DAISYHAT_REPO="daisyhat_Examples" # the github repo to connect to
-# DAISYHAT_USER="TheSlowGrowth" # the github username
-# DAISYHAT_GH_TOKEN="" # the github access token that allows to add/remove runners
\ No newline at end of file
+# export DAISYHAT_DEVICE_MAJOR=189 # major device number of devices to passthrough to the container
+# export DAISYHAT_REPO="daisyhat_Examples" # the github repo to connect to
+# export DAISYHAT_USER="TheSlowGrowth" # the github username
+# export DAISYHAT_GH_TOKEN="" # the github access token that allows to add/remove runners
+
```
Comment (TheSlowGrowth) [inline scripts/setupEnvVars.sh:12]: The `Appended from ...` comment that you had here in one of the comits could actually be added from the setup script. I like it as a reminder to understand what's going on when you look at this file long after the initial setup.

**Review by TheSlowGrowth — COMMENTED** [review]
Thanks for looking into this, I highly appreciate your input!

**Comment by TheSlowGrowth** [comment]
I initially had problems with adding the user to the docker group during the initial setup. There's apparently no way to "apply" the new access rights to a script that's already running. That's why I did the weird `sudo -u $USER docker build -t ${DAISYHAT_DOCKER_IMAGE} .` at the end of the setup script. 
It was a little flaky for me, did it work for you?

File: scripts/raspberryPiSetup.sh (hunk 1 above)
Comment (recursinging) [reply scripts/raspberryPiSetup.sh:49]: `export` makes the variable available not just to the current shell, but all processes started from the current shell.  For me the variables were not available to `startDockerContainer.sh`  causing the `docker run` command to fail.

File: scripts/raspberryPiSetup.sh (hunk 2 above)
Comment (recursinging) [reply scripts/raspberryPiSetup.sh:56]: It works, but it takes precedence over `.profile`, in which `.bashrc` is included by default.  In '.bashrc' are a number of shell aliases and colors which never get sourced if `.bash_profile` is present.  Not really relevant unless you want to shell into the container.

Comment (recursinging) [reply scripts/raspberryPiSetup.sh:56]: _Not really relevant unless you want to shell into the container._  - I take this back... This is relevant when logged into the PI over SSH. Not to the docker container.

File: docker/Dockerfile (hunk 3 above)
Comment (recursinging) [reply docker/Dockerfile:39]: In my case the Dockerfile build failed repeatedly due to this... It's just a bash OR, and shouldn't do anything unless `service udev restart` returns non-0.  I'm also no bash expert, but after removing it the Dockerfile build ran through without issue.

File: scripts/setupEnvVars.sh (hunk 4 above)
Comment (recursinging) [reply scripts/setupEnvVars.sh:12]: `raspberryPiSetup.sh` does append this. It was just still in there after a test run I did.  Those commits were just to clean up.

**Comment by recursinging** [comment]
> It was a little flaky for me, did it work for you?

Yes, it worked for me.  I did need to reboot the PI (or restart my ssh session) before running the `startDockerContainer.sh` script though.

A disclaimer, I put this PR together here in the office this morning, from memory of the issues I ran into yesterday while setting it up on my RPI.  I went through the process twice atop fresh installs of Ubuntu 21.04 on the RPI, but haven't yet had the chance to test these changes directly.  I can do that this evening to verify them, as well as check for any more inconsistencies.

File: scripts/raspberryPiSetup.sh (hunk 1 above)
Comment (TheSlowGrowth) [reply scripts/raspberryPiSetup.sh:49]: weird, it was working for me. Maybe a remainder from some earlier setup attempts. I should wipe my Pi and do the setup again to confirm.
Your explaination sounds very reasonable, thanks for pointing this out

File: scripts/raspberryPiSetup.sh (hunk 2 above)
Comment (TheSlowGrowth) [reply scripts/raspberryPiSetup.sh:56]: Ah I see, I wasn't aware of this. Cool, thanks!

File: docker/Dockerfile (hunk 3 above)
Comment (TheSlowGrowth) [reply docker/Dockerfile:39]: Oh, interesting. Thanks, good to know.

File: scripts/setupEnvVars.sh (hunk 4 above)
Comment (TheSlowGrowth) [reply scripts/setupEnvVars.sh:12]: Ah, my bad :-)

**Comment by TheSlowGrowth** [comment]
Amazing! Thanks a lot. Let me know if your tests succeed and I'll merge this. I won't have time to work on daisyHat today, but I'll try to tackle #8 on the weekend and test your changes in the process.
Also I don't think we have to be too careful right now. If we break things in a PR we'll just fix them again later.

**Comment by recursinging** [comment]
OK, I just tested it against a fresh install.  The script ran through without error.  As mentioned I needed to log out and back in to my ssh session again to run the `startDockerContainer.sh` script.  I don't see a way around this.  I'd say this can be merged as is.

**Pull request merged**